### PR TITLE
Reducers and actions

### PR DESCRIPTION
**Pull request by lazyCode-404**
### The following were implemented:
- Added Redux Toolkit
- Structured the application files using a "feature folder"
- Wrote book's actions and reducer using the ducks pattern

### CODE REVIEW

File: src/components/BookContainer.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        <div key={book.id}>
+          <li className="books-content">
+            {book.title}
+            {' by '}
+            {book.author}
+            {' - '}
+            {book.category}
+          </li>
+          <button type="button" id={book.id}>Remove</button>
+        </div>
```
Comment (RolandM99) [inline src/components/BookContainer.js:20]: - [ ] Kindly, make sure that your `Remove` button to delete a book works as expected. For now, when I click on it, nothing happens. Make sure that you created a function for removing a book and you pass it as the event handler. means you'll use the `onClick{pass your function}`

**Review by RolandM99 — CHANGES_REQUESTED** [review]
Hi @BwalyaNonde , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight 💯 
- Good commit message ✔️ 
- No Linter errors ✔️ 
- Follow Github flow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on @Rolandm99 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookContainer.js (hunk 1 above)
Comment (lazyCode-404) [reply src/components/BookContainer.js:20]: Hi @RolandM99! The project requirement  is  that the remove button is suppose to work in the next project not this on. when adding selectors and dispatch.

**Review by ishimwezachee — APPROVED** [review]
Hi @BwalyaNonde , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Good commit message ✔️
No Linter errors ✔️
Follow Github flow ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**